### DISCUSSION TITLE
Typical [android] workflow

### DISCUSSION THREAD

**Discussion by msloco** [KnugiHK/WhatsApp-Chat-Exporter#92] [Q&A]
I wonder if there is a "typical" workflow we can get inspiration from and contribute to. That is a repeatable process for "offloading" chats and media from a phone to a backup, in a <possibly> incremental fashion, while leaving a possibly controlled part of the data (last <so many months> of data on the phone)
I can contribute I have been successfully copying media, rather much faster than with adb/cable, by using rsync on the PC side and "SSHelper" on the phone acting as a server (I am not endorsing this app, just found it works for me), and I think I might be able to delete backupped media older than a certain date - that is just a matter of working with rsync command link, hopefully. I assume whatsapp will cope with possibly disappearing files from the filesystem.
Another story is the database and its "export" to behave incrementally. Has anyone managed to do so ? or is it possible to run the exporter on two separate .db but pointing for the output to the same directory with some paging strategy so to make the following exports not to overwrite the previous ones ?
please share !
-michele

**Comment by KnugiHK** [comment]
As the project progresses, I’m starting to get a clearer sense of what you're asking for!

> I wonder if there is a "typical" workflow we can get inspiration from and contribute to. That is a repeatable process for "offloading" chats and media from a phone to a backup, in a incremental fashion, while leaving a possibly controlled part of the data (last of data on the phone)
I can contribute I have been successfully copying media, rather much faster than with adb/cable, by using rsync on the PC side and "SSHelper" on the phone acting as a server (I am not endorsing this app, just found it works for me), and I think I might be able to delete backupped media older than a certain date - that is just a matter of working with rsync command link, hopefully. I assume whatsapp will cope with possibly disappearing files from the filesystem.

Definitely! The media folder can be offloaded from the phone and merged whenever needed. Both WhatsApp and this exporter support that. See also: #133.

> Another story is the database and its "export" to behave incrementally. Has anyone managed to do so ? or is it possible to run the exporter on two separate .db but pointing for the output to the same directory with some paging strategy so to make the following exports not to overwrite the previous ones?

I think it’s possible! We could implement a checkpoint file that tracks which message and the previous page number for each chat, allowing future exports to continue from where the last one left off.

**Reply by KnugiHK** [reply]
The incremental exports is tracked in #136.
